### DISCUSSION TITLE
Accessing only the first element on a list of objects without lopping

### DISCUSSION THREAD

**Discussion by tawmoto** [nlohmann/json#3094] [Q&A] [answered]
Hi, I have a list of objects, but in fact I have only a single object there, something like:

```
{
  "animals": {
    "dog": {
      "name": "Bart",
      "age": 20
    }
  }
}
```
I know the first key name, `animals`, I don't know the second key name, `dog`, but I know for sure that it's **only one** element into the list of objects.
What I am doing now:

```
for (const auto& categ : json.at("animals"))
{
    // read name and age
    break;
}
```

Isn't a better solution to avoid the loop, and to access directly the `animals[0]` somehow?

thank you

**Comment by nlohmann** [comment, marked as answer]
[front](https://json.nlohmann.me/api/basic_json/front/) is the function you are looking for.

**Reply by tawmoto** [reply]
> [front](https://json.nlohmann.me/api/basic_json/front/) is the function you are looking for.

Great, thanks a lot Niels!

**Reply by tawmoto** [reply]
hi again @nlohmann, I am sorry to disturb you again, the function `front` works great for the following json
```
{
  "animals": {
    "dog": {
      "name": "Bart",
      "age": 20
    }
  }
}
```
but is there a chance that I can read the `dog` string? I see that it's the key for that object containing `{name, age}` but I don't know how to read it. 

I am doing `json.front()` and I get this json
```
{ "dog": {
      "name": "Bart",
      "age": 20
    }
  }
```
But I need to read that `dog` string, because sometimes it's `cat`.  :-)

Thank you

**Reply by tawmoto** [reply]
I did it with `.begin().key()`, I hope it's correct, sorry for disturbing you. :-)

**Reply by nlohmann** [reply]
Yes, that's the way to go. Again, mind the behavior is undefined if the object is empty.
